// business-experts-server/src/content.js
export const homeContent = {
  stats: [
    { value: "200+", label: "Projects" },
    { value: "10+", label: "Years" },
    { value: "100%", label: "Success" }
  ],
  services: [
    {
      title: "Mandatory Registrations",
      copy: "End-to-end registrations for Pvt Ltd, LLP, and liaison offices.",
      icon: "doc",
      image: "https://images.unsplash.com/photo-1450101499163-c8848c66ca85?auto=format&fit=crop&w=900&q=80"
    },
    {
      title: "Govt. Approvals",
      copy: "Smooth handling of government approvals, tax registrations, and licensing requirements.",
      icon: "check",
      image: "https://images.unsplash.com/photo-1554224155-6726b3ff858f?auto=format&fit=crop&w=900&q=80"
    },
    {
      title: "FDI & Ventures",
      copy: "Expert guidance on FDI entry routes, FEMA/RBI compliance, and structuring joint ventures.",
      icon: "handshake",
      image: "https://images.unsplash.com/photo-1521791136064-7986c2920216?auto=format&fit=crop&w=900&q=80"
    },
    {
      title: "Real Estate",
      copy: "Assistance with industrial land acquisition, commercial leasing, title checks, and due diligence.",
      icon: "land",
      image: "https://images.unsplash.com/photo-1486406146926-c627a92ad1ab?auto=format&fit=crop&w=900&q=80"
    },
    {
      title: "Factory Setup",
      copy: "EPC registration, DPR creation, documentation, product compliance, sourcing, and logistics.",
      icon: "factory",
      image: "https://images.unsplash.com/photo-1513828583688-c52646db42da?auto=format&fit=crop&w=900&q=80"
    },
    {
      title: "Growth Strategy",
      copy: "Growth plans, investor meetings, funding support, strategic partnerships, and advisory models.",
      icon: "growth",
      image: "https://images.unsplash.com/photo-1556761175-b413da4baf72?auto=format&fit=crop&w=900&q=80"
    }
  ]
};
